js-pkg/tests/src/driftdb.test.ts

```diff
@@ -39,18 +38,17 @@ class CallbackExpecter<T> {
         if (this.resolve) {
             this.resolve(value);
             this.resolve = null;
-            this.reject = null;
         } else {
             this.nextValue = value;
         }
     }
 }
 
-async function connectToNewRoom(): Promise<{ db: DbConnection, room: RoomResult }> {
+async function connectToNewRoom({binary}: {binary?: boolean} = {}): Promise<{ db: DbConnection, room: RoomResult }> {
     let api = new Api(API_SERVER);
     let room = await api.newRoom();
     let db = new DbConnection();
-    await db.connect(room.socket_url);
+    await db.connect(room.socket_url, binary);
```

Comment:
As far as I understand it, `db.send()` takes some JavaScript values whether or not `binary=true`. The only difference is that if your JavaScript values include TypedArrays or an ArrayBuffer, you need to set `binary=true` for it to work. So what's the point of having `binary=false`? Does it make sense to support JSON at all?

And if there is some use case where developers need to know about the transport format of the messages (and it's not just some implementation detail), then might they care about what format is actually being used, i.e. this isn't just binary, it's specifically CBOR.

I guess the TLDR is: Maybe we should just always use CBOR for serialization and not give the developer any options for the transport format. _Or_ if we _do_ think developers should be able to control the transport format, then maybe the options should be JSON vs CBOR (possibly with CBOR being the default?).